src/source/mod.rs

```diff
@@ -85,6 +85,26 @@ mod triangle;
 mod uniform;
 mod zero;
 
+#[cfg(feature = "dither")]
+#[cfg_attr(docsrs, doc(cfg(feature = "dither")))]
+mod dither;
+#[cfg(feature = "dither")]
+#[cfg_attr(docsrs, doc(cfg(feature = "dither")))]
+pub use self::dither::{Algorithm as DitherAlgorithm, Dither};
+
+/// Creates a dithered source using the specified algorithm.
+///
+/// Dithering eliminates quantization artifacts during digital audio playback
+/// and when converting between bit depths. Apply at the target output bit depth.
+#[cfg(feature = "dither")]
+#[cfg_attr(docsrs, doc(cfg(feature = "dither")))]
+pub fn dither<I>(input: I, target_bits: BitDepth, algorithm: DitherAlgorithm) -> Dither<I>
```

Comment:
replace this with a default implementation on source? So you can do `sine().dither().amplify().pausable()` etc.

Reply:
otherwise good to merge btw